raiden-ts/src/transfers/epics/locked.ts

```diff
@@ -69,12 +68,12 @@ import { Direction } from '../state';
  * @param balanceProof - Balance proof to increase nonce from
  * @returns Increased nonce, or One if no balance proof provided
  */
-function nextNonce(balanceProof?: SignedBalanceProof): UInt<8> {
-  return (balanceProof?.nonce ?? Zero).add(1) as UInt<8>;
+function nextNonce(balanceProof: BalanceProof): UInt<8> {
+  return balanceProof.nonce.add(1) as UInt<8>;
 }
 
-function getChannelLocksroot(channel: Channel, secrethash: Hash): Hash {
-  const locks: Lock[] = (channel.own.locks ?? []).filter((l) => l.secrethash !== secrethash);
+function withoutLock(end: ChannelEnd, secrethash: Hash): Hash {
```

Comment:
Any reason for the change of the name? At first glance it's not obvious to me what the function does if I don't see the implementation.

Reply:
It was used because of the optional `locks`, but also was only useful for our `own` end (because previously we didn't receive, so there was no need to calculate partner's locks without a specific lock). I expanded it to be used for both ends. The name was not clear before, `getChannelLocksroot` looks like `getLocksroot`, which just calculates the hash of the locks. Now, `withoutLock(channel.partner, secrethash)` will return the locksroot of `channel.partner` *without lock* `secrethash`.

Reply:
It could be `getEndLocksrootWithoutLock`, but I figured out we didn't need this much for an internal helper function :smile: